numpy/core/setup.py

```diff
@@ -737,11 +738,13 @@ def get_mathlib_info(*args):
         ):
             is_cpp = lang == 'c++'
             if is_cpp:
-                # this a workround to get rid of invalid c++ flags
+                # this a workaround to get rid of invalid c++ flags
                 # without doing big changes to config.
                 # c tested first, compiler should be here
                 bk_c = config_cmd.compiler
                 config_cmd.compiler = bk_c.cxx_compiler()
+                config_cmd.compiler.compiler.extend(NPY_CXX_FLAGS)
+                config_cmd.compiler.compiler_so.extend(NPY_CXX_FLAGS)
```

Comment:
This needs some special casing to make sure `compiler.compiler exists`. It is [failing on windows](https://dev.azure.com/numpy/numpy/_build/results?buildId=23477&view=logs&j=f758ecd3-9bca-5197-a703-541c3e314cdb&t=5b103f5c-b1f2-5e5c-a607-e45c0963e9aa&l=252) with
```
AttributeError: 'MSVCCompiler' object has no attribute 'compiler'. Did you mean: compile?
```